### PR TITLE
storagenodedb: add trigger table for calculating piece sums

### PR DESCRIPTION
**Pull request by zeebo**
What: Use triggers to keep track of pieceinfo sums

Why: This eliminates the slow startup problems of the cache

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact: Should be minimal outside of startup situations

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: internal/dbutil/dbschema/schema.go [hunk 1]
```diff
@@ -111,6 +119,9 @@ func (schema *Schema) Sort() {
 	sort.Slice(schema.Indexes, func(i, k int) bool {
 		return schema.Indexes[i].Name < schema.Indexes[k].Name
 	})
+	sort.Slice(schema.Triggers, func(i, k int) bool {
```
Comment (egonelbre) [inline internal/dbutil/dbschema/schema.go]: Let's add new testdata in `query_test.go`

File: storagenode/storagenodedb/infodb.go [hunk 2]
```diff
@@ -70,8 +68,8 @@ func NewInfoInMemory() (*InfoDB, error) {
 		}))
 
 	infoDb := &InfoDB{db: db}
-	infoDb.pieceinfo = pieceinfo{InfoDB: infoDb, space: spaceUsed{used: 0, once: sync.Once{}}}
-	infoDb.bandwidthdb = bandwidthdb{InfoDB: infoDb, bandwidth: bandwidthUsed{used: 0, mu: sync.RWMutex{}, usedSince: time.Time{}}}
+	infoDb.pieceinfo = pieceinfo{InfoDB: infoDb}
+	infoDb.bandwidthdb = bandwidthdb{InfoDB: infoDb}
```
Comment (egonelbre) [inline storagenode/storagenodedb/infodb.go]: we now also won't need these fields

File: storagenode/storagenodedb/infodb.go [hunk 3]
```diff
@@ -38,11 +36,9 @@ func newInfo(path string) (*InfoDB, error) {
 	if err != nil {
 		return nil, ErrInfo.Wrap(err)
 	}
-
 	dbutil.Configure(db, mon)
 
 	infoDb := &InfoDB{db: db}
-	infoDb.pieceinfo = pieceinfo{InfoDB: infoDb}
 	infoDb.bandwidthdb = bandwidthdb{InfoDB: infoDb}
```
Comment (egonelbre) [inline storagenode/storagenodedb/infodb.go]: this also

Comment (zeebo) [reply storagenode/storagenodedb/infodb.go]: The bandwidthdb is still doing the caching stuff, and so the infodb still needs to be returning the same one every time. I just removed the text zero initializing a bunch of fields to their zero values.

**Pull request closed without merging**